impl/src/main/java/edu/si/trellis/CassandraResourceService.java

```diff
@@ -104,17 +109,23 @@ public CassandraResourceService(Delete delete, Get get, ImmutableInsert immutabl
      */
     @PostConstruct
     void initializeRoot() {
-
-        IRI rootIri = TrellisUtils.getInstance().createIRI(TRELLIS_DATA_PREFIX);
-        try {
-            if (get(rootIri).toCompletableFuture().get().equals(MISSING_RESOURCE)) {
-                Metadata rootResource = builder(rootIri).interactionModel(BasicContainer).build();
-                create(rootResource, null).toCompletableFuture().get();
+    	final RDF rdf = getInstance();
+        final IRI root = rdf.createIRI(TRELLIS_DATA_PREFIX);
+        final IRI rootAuth = rdf.createIRI(TRELLIS_DATA_PREFIX + "#auth");
+        try (final Dataset dataset = rdf.createDataset()) {
+            dataset.add(rdf.createQuad(Trellis.PreferAccessControl, rootAuth, ACL.mode, ACL.Read));
```

Comment:
I'm not happy about this-- I would much rather leave WebAC to WebAC. Perhaps we can take a little more time to delve into why the various `@PostConstruct`s aren't doing what they should? We will need to solve this anyway to move onto a µprofile platform.

Reply:
Initialization seems to be a hard problem. I think the lack of a hasACL implementation may have been connected to the PostConstruct issue, at least it seems possible that TrellisHttpResource was not seeing the ACL property accurately and that may have caused it's ACL part to be skipped. Ideally of course the WebACService itself would set the root ACL.. but this would have to follow root creation at the trellis level. Interesting problem.

Reply:
Okay, so it's a dependency on order? One must follow the other?

Reply:
Yes, if one is to set up the root object (if not existing), and another is going to add a default root ACL (if not existing).

Reply:
Got it. Okay, it's actually `trellis-cassandra` that sets up the root, because each backend does that differently. We may need to abstract over and make that explicit and provide a method for `trellis-http` to call to initialize any backend. @acoburn, did we have a reason not to do that in the past?